packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -154,85 +160,92 @@ const sharedStyles = StyleSheet.create({
         zIndex: 1,
-        transition: "0.15s ease-in-out",
-        transitionProperty: "transform, color",
+        transition: "transform 0.15s ease-in-out",
     },
 });
 
 const styles: Record<string, any> = {};
 const _generateStyles = (
     checked: boolean,
-    disabled: boolean,
     clickable: boolean,
+    disabled: boolean,
+    theme: SwitchThemeContract,
+    themeName: string,
 ) => {
-    const checkedStyle = `${checked}-${disabled}-${clickable}`;
+    const checkedStyle = `${checked}-${clickable}-${disabled}-${themeName}`;
     // The styles are cached to avoid creating a new object on every render.
     if (styles[checkedStyle]) {
         return styles[checkedStyle];
     }
 
     let newStyles: Record<string, any> = {};
-
-    const disabledBlue = mix(Color.blue, Color.offBlack50);
-    const activeBlue = mix(Color.offBlack32, Color.blue);
+    const sharedSwitchStyles = {
+        cursor: clickable ? "pointer" : "auto",
+        ":hover": {
+            outline: clickable
+                ? `solid ${theme.size.width.small}px ${theme.color.outline.default}`
+                : "none",
+        },
+    };
 
     if (checked) {
         newStyles = {
             switch: {
-                backgroundColor: disabled ? disabledBlue : Color.blue,
+                backgroundColor: disabled
+                    ? theme.color.bg.switch.disabledOn
+                    : theme.color.bg.switch.on,
                 ":active": {
-                    backgroundColor: !disabled && clickable && activeBlue,
-                },
-                ":focus-within": {
-                    outline: `solid ${Spacing.xxxxSmall_2}px ${Color.blue}`,
-                    outlineOffset: 1,
-                },
-                ":hover": {
-                    outline: clickable
-                        ? `solid ${Spacing.xxxxSmall_2}px ${Color.blue}`
-                        : "none",
+                    backgroundColor:
+                        !disabled &&
+                        clickable &&
+                        theme.color.bg.switch.activeOn,
                 },
+                ...sharedSwitchStyles,
             },
             slider: {
-                transform: `translateX(${Spacing.medium_16}px)`,
+                transform: theme.spacing.transform,
             },
             icon: {
-                color: disabled ? disabledBlue : Color.blue,
-                transform: `translateX(${Spacing.medium_16}px)`,
+                color: disabled
+                    ? theme.color.bg.icon.disabledOn
+                    : theme.color.bg.icon.on,
+                transform: theme.spacing.transform,
             },
         };
     } else {
         newStyles = {
             switch: {
-                backgroundColor: disabled ? Color.offBlack32 : Color.offBlack50,
+                backgroundColor: disabled
+                    ? theme.color.bg.switch.disabledOff
+                    : theme.color.bg.switch.off,
```

Comment:
**praise:** Awesome! I love who these tokens are super readable and map to their elements.